python/ctsm/crop_calendars/regrid_ggcmi_shdates.py

```diff
@@ -32,47 +33,106 @@ def define_arguments(parser):
     parser.add_argument(
         "-rt",
         "--regrid-template-file",
-        help="Template file to be used in regridding of inputs.",
+        help="Template netCDF file to be used in regridding of inputs. This can be a CLM output file (i.e., something with 1-d lat and lon variables) or a CLM surface dataset (i.e., something with 2-d LATIXY and LONGXY variables).",
         type=str,
         required=True,
     )
     return parser
 
 
+# lat or lon
+def import_coord_1d(ds, coordName):
+    da = ds[coordName]
+    if len(da.dims) != 1:
+        raise RuntimeError(f"Expected 1 dimension for {coordName}; found {len(da.dims)}: {da.dims}")
+    return da, len(da)
+
+
+# LATIXY or LONGXY
+def import_coord_2d(ds, coordName, varName):
+    da = ds[varName]
+    thisDim = [x for x in da.dims if coordName in x]
+    if len(thisDim) != 1:
+        raise RuntimeError(
+            f"Expected 1 dimension name containing {coordName}; found {len(otherDim)}: {otherDim}"
+        )
+    thisDim = thisDim[0]
+    otherDim = [x for x in da.dims if coordName not in x]
+    if len(otherDim) != 1:
+        raise RuntimeError(
+            f"Expected 1 dimension name not containing {coordName}; found {len(otherDim)}: {otherDim}"
+        )
+    otherDim = otherDim[0]
+    da = da.astype(np.float32)
+    da = da.isel({otherDim: [0]}).squeeze().rename({thisDim: coordName}).rename(coordName)
+    da = da.assign_coords({coordName: da.values})
+    da.attrs["long_name"] = "coordinate " + da.attrs["long_name"]
+    da.attrs["units"] = da.attrs["units"].replace(" ", "_")
+    return da, len(da)
+
+
 def main(
     regrid_resolution, regrid_template_file_in, regrid_input_directory, regrid_output_directory
 ):
+    print(f"Regridding GGCMI crop calendars to {regrid_resolution}:")
 
-    # Ensure we can call necessary shell scripts
-    for cmd in ["ncks", "ncrename", "ncpdq", "cdo"]:
+    # Ensure we can call necessary shell script(s)
+    for cmd in ["cdo"]:
         run_and_check(f"{cmd} --help")
```

Comment:
This is an interesting and nice feature to check binaries that are needed. I like this.